pythonPackages.pywinrm: 0.1.1 -> 0.2.2

elasticdog opened the pull request:
###### Motivation for this change

Bump version to latest upstream release and remove the already broken/optional Kerberos support.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [x] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

The latest version of *pywinrm* has a new dependency on *requests_ntlm* which requires the lower-level *ntlm-auth*. I've removed the optional *kerberos* dependency, as it was already broken, now requires additional dependencies to function, and I couldn't figure out a clean way of manipulating the `LD_LIBRARY_PATH` (or patch things with the absolute path) to fix the original error:

> error: libgssapi_krb5.so: cannot open shared object file: No such file or directory

The other changed dependencies were just making things accurate with the upstream requirements. Here's the `pipdeptree` output, if you're curious:

```
pywinrm==0.2.2
  - requests [required: >=2.9.1, installed: 2.14.2]
  - requests-ntlm [required: >=0.3.0, installed: 1.0.0]
    - ntlm-auth [required: >=1.0.2, installed: 1.0.3]
      - six [required: Any, installed: 1.10.0]
    - requests [required: >=2.0.0, installed: 2.14.2]
  - six [required: Any, installed: 1.10.0]
  - xmltodict [required: Any, installed: 0.11.0]
```

The tests I disabled require a local IIS server and network access.

elasticdog committed:
pythonPackages.ntlm-auth: init at 1.0.3

elasticdog committed:
pythonPackages.requests_ntlm: init at 1.0.0

elasticdog committed:
pythonPackages.pywinrm: 0.1.1 -> 0.2.2

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -15552,6 +15552,32 @@ in {
+      sha256 = "12cwv0ywlfvspnpjlc49yccnzp7ijvjwrzhl31wzb0y4kqgzzxx0";
+    };
+
+    buildInputs = with self; [ mock pytest unittest2 ];
```
checkInputs

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -21091,22 +21117,25 @@ in {
     };
 
-    propagatedBuildInputs = with self; [ isodate kerberos xmltodict ];
+    buildInputs = with self; [ mock pytest ];
```
checkInputs

FRidh reviewed (changes requested):
Please see the header of the file and move the expressions.

FRidh commented:
> The latest version of pywinrm has a new dependency on requests_ntlm which requires the lower-level ntlm-auth. I've removed the optional kerberos dependency, as it was already broken, now requires additional dependencies to function, and I couldn't figure out a clean way of manipulating the LD_LIBRARY_PATH (or patch things with the absolute path) to fix the original error:

Typically it tries to load the library somewhere with ctypes or cffi. In that function, don't let it use the name of the library but instead pass in the absolute path to the library.

elasticdog commented:
Thank you for the feedback! I force pushed the branch after added the following changes:

* moved the three libraries into dedicated files under `pkgs/development/python-modules/`
* switched to using the proper `checkInputs`
* instead of `fetchurl`, refactored to use `fetchpypi` when possible and `fetchFromGitHub` for *ntlm-auth* since they strip out the tests directory in the PyPI tarball

elasticdog commented:
@FRidh were you wanting any other changes here? Everything is working properly for me when using it locally.

I don't have any real experience with the Kerberos ecosystem, and I did try to trigger the runtime error with the old pywinrm package version (which is why it got labeled as broken) by following the basic instructions laid out in https://github.com/diyan/pywinrm/issues/21, but couldn't reproduce the problem. If you want, I can try to add back in the kerberos support as an optional install and package up the new `requests-kerberos` dependency, but again, I'm not confident that I can test it properly to know if it's working.

FRidh reviewed (approved):
Haven't tested it but looks good to me.
